[wtl] Error when including atlribbon.h

****

I'm trying to use the wtl package, but as soon as I `#include <wtl/atlribbon.h>`, the build fails:

    ...\vcpkg_installed\x86-windows\x86-windows\include\wtl\atlribbon.h(30,10): fatal error C1083: Cannot open include file: 'atlmisc.h': No such file or directory

The error comes from the atlribbon.h file in the WTL package, and from the fact that vcpkg sets the include path to the folder containing the "wtl" folder, not the folder itself. Therefore, I also had to change my includes from `#include <atlribbon.h>` to `#include <wtl/atlribbon.h>` in the first place. WTL just assumes that its include folder is directly in the path.

Easy steps to reproduce:
- Install the WTL package into VS 2019 (I didn't try VS 2022)
- create a new WTL project, any type
- add a vcpkg.json that references the "wtl" package
- switch on "Use Vcpkg Manifest"
- compile

**Proposed solution**

The solution would be to change the include path for the WTL package to directly include the "wtl" folder in the package, not the parent folder of that. After all, the WTL project wizard creates projects with this in mind and surely there are WTL projects out there who use that convention for decade(s). I'm not sure if this breaks some vcpkg rules. At the moment, the package is useless.

**Describe alternatives you've considered**

I can manually add an "Additional Include Directory" in the Visual Studio C/C++ properties, as a workaround:

    $(VcpkgManifestRoot)\vcpkg_installed\$(VcpkgTriplet)\$(VcpkgTriplet)\include\wtl

Note that this is also needed to be added to the Resource Compiler settings, as the .rc files may also include <atlres.h>, which is also a part of WTL.

--- a/ports/wtl/portfile.cmake
+++ b/ports/wtl/portfile.cmake
@@ -14,4 +14,6 @@ file(INSTALL ${SOURCE_PATH}/Include/ DESTINATION ${CURRENT_PACKAGES_DIR}/include
 file(COPY ${SOURCE_PATH}/Samples DESTINATION ${CURRENT_PACKAGES_DIR}/share/${PORT})
 file(COPY ${SOURCE_PATH}/AppWizard DESTINATION ${CURRENT_PACKAGES_DIR}/share/${PORT})
 
+vcpkg_replace_string("${CURRENT_PACKAGES_DIR}/include/wtl/atlribbon.h" "#include <atl" "#include <wtl/atl")
+
 file(INSTALL ${SOURCE_PATH}/MS-PL.txt DESTINATION ${CURRENT_PACKAGES_DIR}/share/${PORT} RENAME copyright)

--- a/ports/wtl/portfile.cmake
+++ b/ports/wtl/portfile.cmake
@@ -10,10 +10,8 @@ vcpkg_from_sourceforge(
         atlmisc.h-bug329.patch
 )
 
-file(INSTALL "${SOURCE_PATH}/Include/" DESTINATION "${CURRENT_PACKAGES_DIR}/include/${PORT}" FILES_MATCHING PATTERN "*.h")
+file(INSTALL "${SOURCE_PATH}/Include/" DESTINATION "${CURRENT_PACKAGES_DIR}/include" FILES_MATCHING PATTERN "*.h")
 file(COPY "${SOURCE_PATH}/Samples" DESTINATION "${CURRENT_PACKAGES_DIR}/share/${PORT}")
 file(COPY "${SOURCE_PATH}/AppWizard" DESTINATION "${CURRENT_PACKAGES_DIR}/share/${PORT}")
-
-vcpkg_replace_string("${CURRENT_PACKAGES_DIR}/include/wtl/atlribbon.h" "#include <atl" "#include <wtl/atl")
 
 file(INSTALL "${SOURCE_PATH}/MS-PL.txt" DESTINATION "${CURRENT_PACKAGES_DIR}/share/${PORT}" RENAME copyright)

--- a/ports/wtl/portfile.cmake
+++ b/ports/wtl/portfile.cmake
@@ -4,16 +4,16 @@ vcpkg_from_sourceforge(
     REF WTL%2010.0.10320%20Release
     FILENAME "WTL10_10320_Release.zip"
     NO_REMOVE_ONE_LEVEL
-	SHA512 086a6cf6a49a4318a8c519136ba6019ded7aa7f2c1d85f78c30b21183654537b3428a400a64fcdacba3a7a10a9ef05137b6f2119f59594da300d55f9ebfb1309
-	PATCHES
-		appwizard_setup.js-vs2022.patch
-		atlmisc.h-bug329.patch
+    SHA512 086a6cf6a49a4318a8c519136ba6019ded7aa7f2c1d85f78c30b21183654537b3428a400a64fcdacba3a7a10a9ef05137b6f2119f59594da300d55f9ebfb1309
+    PATCHES
+        appwizard_setup.js-vs2022.patch
+        atlmisc.h-bug329.patch
 )
 
-file(INSTALL ${SOURCE_PATH}/Include/ DESTINATION ${CURRENT_PACKAGES_DIR}/include/${PORT} FILES_MATCHING PATTERN "*.h")
-file(COPY ${SOURCE_PATH}/Samples DESTINATION ${CURRENT_PACKAGES_DIR}/share/${PORT})
-file(COPY ${SOURCE_PATH}/AppWizard DESTINATION ${CURRENT_PACKAGES_DIR}/share/${PORT})
+file(INSTALL "${SOURCE_PATH}/Include/" DESTINATION "${CURRENT_PACKAGES_DIR}/include/${PORT}" FILES_MATCHING PATTERN "*.h")
+file(COPY "${SOURCE_PATH}/Samples" DESTINATION "${CURRENT_PACKAGES_DIR}/share/${PORT}")
+file(COPY "${SOURCE_PATH}/AppWizard" DESTINATION $"{CURRENT_PACKAGES_DIR}/share/${PORT}")
 
 vcpkg_replace_string("${CURRENT_PACKAGES_DIR}/include/wtl/atlribbon.h" "#include <atl" "#include <wtl/atl")
 
-file(INSTALL ${SOURCE_PATH}/MS-PL.txt DESTINATION ${CURRENT_PACKAGES_DIR}/share/${PORT} RENAME copyright)
+file(INSTALL "${SOURCE_PATH}/MS-PL.txt" DESTINATION "${CURRENT_PACKAGES_DIR}/share/${PORT}" RENAME copyright)

--- a/ports/wtl/portfile.cmake
+++ b/ports/wtl/portfile.cmake
@@ -12,7 +12,7 @@ vcpkg_from_sourceforge(
 
 file(INSTALL "${SOURCE_PATH}/Include/" DESTINATION "${CURRENT_PACKAGES_DIR}/include/${PORT}" FILES_MATCHING PATTERN "*.h")
 file(COPY "${SOURCE_PATH}/Samples" DESTINATION "${CURRENT_PACKAGES_DIR}/share/${PORT}")
-file(COPY "${SOURCE_PATH}/AppWizard" DESTINATION $"{CURRENT_PACKAGES_DIR}/share/${PORT}")
+file(COPY "${SOURCE_PATH}/AppWizard" DESTINATION "${CURRENT_PACKAGES_DIR}/share/${PORT}")
 
 vcpkg_replace_string("${CURRENT_PACKAGES_DIR}/include/wtl/atlribbon.h" "#include <atl" "#include <wtl/atl")
 

--- a/ports/wtl/vcpkg.json
+++ b/ports/wtl/vcpkg.json
@@ -1,7 +1,7 @@
 {
   "name": "wtl",
   "version-string": "10.0.10320",
-  "port-version": 2,
+  "port-version": 3,
   "description": "Windows Template Library (WTL) is a C++ library for developing Windows applications and UI components.",
   "homepage": "https://sourceforge.net/projects/wtl/"
 }

--- a/ports/wtl/vcpkg.json
+++ b/ports/wtl/vcpkg.json
@@ -3,5 +3,6 @@
   "version": "10.0.10320",
   "port-version": 3,
   "description": "Windows Template Library (WTL) is a C++ library for developing Windows applications and UI components.",
-  "homepage": "https://sourceforge.net/projects/wtl/"
+  "homepage": "https://sourceforge.net/projects/wtl/",
+  "license": "MS-PL"
 }

--- a/ports/wtl/vcpkg.json
+++ b/ports/wtl/vcpkg.json
@@ -1,6 +1,6 @@
 {
   "name": "wtl",
-  "version-string": "10.0.10320",
+  "version": "10.0.10320",
   "port-version": 3,
   "description": "Windows Template Library (WTL) is a C++ library for developing Windows applications and UI components.",
   "homepage": "https://sourceforge.net/projects/wtl/"

--- a/versions/baseline.json
+++ b/versions/baseline.json
@@ -2742,7 +2742,7 @@
     },
     "hareflow": {
       "baseline": "0.1.0",
-      "port-version": 0      
+      "port-version": 0
     },
     "harfbuzz": {
       "baseline": "4.2.0",
@@ -7578,7 +7578,7 @@
     },
     "wtl": {
       "baseline": "10.0.10320",
-      "port-version": 2
+      "port-version": 3
     },
     "wxchartdir": {
       "baseline": "2.0.0",

--- a/versions/w-/wtl.json
+++ b/versions/w-/wtl.json
@@ -1,7 +1,7 @@
 {
   "versions": [
     {
-      "git-tree": "b1b8e94e75ab0aecd6fa3814c161640851d35606",
+      "git-tree": "1edf05e155b3d36393b83243f61de20b8cc25a4d",
       "version": "10.0.10320",
       "port-version": 3
     },

--- a/versions/w-/wtl.json
+++ b/versions/w-/wtl.json
@@ -1,7 +1,7 @@
 {
   "versions": [
     {
-      "git-tree": "c53f87dbe7c641ec664148035d99f1e2f5d7c7f8",
+      "git-tree": "77302be4bab5f3221389c7948462e15ac941002b",
       "version": "10.0.10320",
       "port-version": 3
     },

--- a/versions/w-/wtl.json
+++ b/versions/w-/wtl.json
@@ -1,6 +1,11 @@
 {
   "versions": [
     {
+      "git-tree": "b1b8e94e75ab0aecd6fa3814c161640851d35606",
+      "version": "10.0.10320",
+      "port-version": 3
+    },
+    {
       "git-tree": "3345336300f47e924a80d3be0f3fb76c558a54fe",
       "version-string": "10.0.10320",
       "port-version": 2

--- a/versions/w-/wtl.json
+++ b/versions/w-/wtl.json
@@ -1,7 +1,7 @@
 {
   "versions": [
     {
-      "git-tree": "1edf05e155b3d36393b83243f61de20b8cc25a4d",
+      "git-tree": "c53f87dbe7c641ec664148035d99f1e2f5d7c7f8",
       "version": "10.0.10320",
       "port-version": 3
     },

